kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/OppslagstjenesteClient.java

```diff
@@ -1,67 +1,96 @@
+    public KontaktInfo hentKontaktInformasjon(LookupParameters lookupParameters) {
         final HentPersonerForespoersel hentPersonerForespoersel = HentPersonerForespoersel.builder()
                 .addInformasjonsbehov(Informasjonsbehov.KONTAKTINFO, Informasjonsbehov.SIKKER_DIGITAL_POST, Informasjonsbehov.SERTIFIKAT, Informasjonsbehov.VARSLINGS_STATUS)
-                .addPersonidentifikator(pid)
+                .addPersonidentifikator(lookupParameters.getIdentifier())
                 .build();
 
         WebServiceTemplate template = createWebServiceTemplate(HentPersonerRespons.class.getPackage().getName());
 
-        final HentPersonerRespons hentPersonerRespons = (HentPersonerRespons) template.marshalSendAndReceive(conf.url, hentPersonerForespoersel);
+        WebServiceMessageCallback callback = conf.isPaaVegneAvEnabled() ? noopCallback -> { } : addPaaVegneAvToSoapHeader(lookupParameters.getClientOrgnr());
+        final HentPersonerRespons hentPersonerRespons = (HentPersonerRespons) template.marshalSendAndReceive(conf.url, hentPersonerForespoersel, callback);
 
         return KontaktInfo.from(hentPersonerRespons);
 
     }
 
+    private WebServiceMessageCallback addPaaVegneAvToSoapHeader(String orgnumber) {
+        Oppslagstjenesten oppslagstjenesten = new Oppslagstjenesten();
+        oppslagstjenesten.setPaaVegneAv(orgnumber);
+        return webServiceMessage -> {
+            SoapMessage soapMessage = (SoapMessage) webServiceMessage;
+            SoapHeader soapHeader = soapMessage.getSoapHeader();
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            final Transformer transformer = transformerFactory.newTransformer();
+            try {
+                JAXBSource jaxbSource = new JAXBSource(JAXBContext.newInstance(Oppslagstjenesten.class.getPackage().getName()), oppslagstjenesten);
+                transformer.transform(jaxbSource, soapHeader.getResult());
+            } catch (JAXBException e) {
+                String m = String.format("Failed to add paa vegne av oppslag for %s", orgnumber);
+                throw new OppslagstjenesteException(m, e);
+            }
+        };
+    }
+
     /**
      *  Tjenesten kan brukes for å sende forsendelser av brev til mottakere som har reservert seg eller ikke registrert/oppdatert sin kontaktinformasjon.
      */
-    public PrintProviderDetails getPrintProviderDetails() {
+    public PrintProviderDetails getPrintProviderDetails(String orgnumber) {
         HentPrintSertifikatForespoersel request = new HentPrintSertifikatForespoersel();
         WebServiceTemplate template = createWebServiceTemplate(HentPrintSertifikatRespons.class.getPackage().getName());
 
-        HentPrintSertifikatRespons response = (HentPrintSertifikatRespons) template.marshalSendAndReceive(conf.url, request);
+        WebServiceMessageCallback callback = conf.isPaaVegneAvEnabled() ? emptyCallback -> { } : addPaaVegneAvToSoapHeader(orgnumber);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00108)